src/elexmodel/handlers/data/CombinedData.py

```diff
@@ -74,29 +86,41 @@ def get_reporting_units(self, percent_reporting_threshold, turnout_factor_lower,
         reporting_units["reporting"] = int(1)
         reporting_units["expected"] = True
 
-        return reporting_units
-
-    def get_nonreporting_units(self, percent_reporting_threshold, turnout_factor_lower, turnout_factor_upper):
-        """
-        Get nonreporting data. These are units where expected vote is less than the percent reporting threshold
-        """
         # units where expected vote is less than the percent reporting threshold
         nonreporting_units = self.data[self.data.percent_expected_vote < percent_reporting_threshold].reset_index(
             drop=True
         )
 
-        # remove unexpected units
-        unexpected_units = self._get_unexpected_units(
-            percent_reporting_threshold, turnout_factor_lower, turnout_factor_upper
-        )
+        # remove unexpected and non-predictive units
         nonreporting_units = nonreporting_units[
             ~nonreporting_units.geographic_unit_fips.isin(unexpected_units.geographic_unit_fips)
         ].reset_index(drop=True)
+        nonreporting_units = nonreporting_units[
+            ~nonreporting_units.geographic_unit_fips.isin(non_predictive_units.geographic_unit_fips)
+        ].reset_index(drop=True)
 
         nonreporting_units["reporting"] = int(0)
         nonreporting_units["expected"] = True
 
-        return nonreporting_units
+        # finalize all unexpected/non-predictive units
+        unexpected_units["predictive"] = True
+        non_predictive_units["predictive"] = False
+        all_unexpected_units = pd.concat([unexpected_units, non_predictive_units]).reset_index(drop=True)
+        # since we were not expecting them, we have don't have their county or district
+        # from preprocessed data. so we have to add that back in.
+        if "county_fips" in aggregates:
+            all_unexpected_units["county_fips"] = all_unexpected_units["geographic_unit_fips"].apply(
+                self._get_county_fips_from_geographic_unit_fips
+            )
+        if "district" in aggregates:
+            all_unexpected_units["district"] = all_unexpected_units["geographic_unit_fips"].apply(
+                self._get_district_from_geographic_unit_fips
+            )
+
+        all_unexpected_units["reporting"] = int(0)
+        all_unexpected_units["expected"] = False
```

Comment:
how about, instead of having multiple strings here that do slightly different things. we have one field called `unit_category` which is either `reporting` or `non_reporting` or `unexpected` or `non_modeled`?

Reply:
relatedly, I think we should confirm that this column would also be passed to s3 so that we know what category each unit is in (and maybe also returned by the model)?

Reply:
> how about, instead of having multiple strings here that do slightly different things. we have one field called `unit_category` which is either `reporting` or `non_reporting` or `unexpected` or `non_modeled`?

This sounds good and I just pushed changes for this, so let me know what you think.  I didn't get rid of the `reporting` column, though, because I know that one gets aggregated up to a sum total of units that are reporting within an aggregate (`county_fips`, `postal_code`, etc.) 🤔

Reply:
> relatedly, I think we should confirm that this column would also be passed to s3 so that we know what category each unit is in (and maybe also returned by the model)?

Not currently, unless I'm missing it somewhere.  Let me see, I should be able to make that happen 🤔

Reply:
> > relatedly, I think we should confirm that this column would also be passed to s3 so that we know what category each unit is in (and maybe also returned by the model)?
> 
> Not currently, unless I'm missing it somewhere. Let me see, I should be able to make that happen 🤔

Done!!  Take a look; I'm merging the `unit_category` column into the `ModelResults.final_results["unit_data"]`, so LMK if there's a better way to do it 🎉